app/carbonapi/app.go

```diff
@@ -529,7 +550,10 @@ func deferredAccessLogging(r *http.Request, accessLogDetails *carbonapipb.Access
 		accessLogger.Info("request served", zap.Any("data", *accessLogDetails))
 		apiMetrics.Responses.Add(1)
 	}
-	prometheusMetrics.Responses.WithLabelValues(fmt.Sprintf("%d", accessLogDetails.HttpCode), accessLogDetails.Handler).Inc()
+
+	if app != nil {
```

Comment:
You can't call method for struct App, if App eq nil. I think this condition can be rid.

Reply:
I can, and it actually happened during tests. I have added this line after a test failed because of `app == nil`.